Add new Reactions UML to master

joeyfreeman opened the pull request:
UML showing Reaction hierarchy

joeyfreeman committed:
Added Reaction classes

joeyfreeman committed:
Fixed .pro file

Debug build now compiles correctly

joeyfreeman committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into master_reaction

joeyfreeman committed:
Added review and reply tests, updated testmain

-Added test for creation of review and reply
-Prevented direct instantiation of Reaction base class
-Updated testmain to make overall test result more obvious
-Changed window text of login page to "Login"

joeyfreeman committed:
Added UML for Reactions

Changed uml_review_media to uml_reactions

joeyfreeman committed:
Removed uml_review_media.qmodel

joeyfreeman committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into master_reaction

allanhaywood reviewed (changes requested):
Update UML to also indicate the "has a" relationships.

Example. If the episodes are the classes that have an array of reactions and comments, that should be indicated in the UML. NOTE: I don't believe the qmodel editor allows a class to reference itself, so just put an annotation about how a reply can have 0 to many replies.

Move the methods out of DatabaseManager and onto the classes that own the content. Also, we shouldn't have to reference IDs. The Review, Comment, and Reply classes, would have a "PostReply" method. If that makes sense.

I plan to be on campus after class on Monday, so we can get together and discuss further.

joeyfreeman committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into master_reaction

joeyfreeman committed:
Updated Reactions classes and UML

-Replies can not be added to replies
-Changed userName to userID
-Moved reply vector and addReply() method from Reaction to Review and
Comment subclasses
-Updated tests to work with changes
-Updated UML to reflect changes and relationships

joeyfreeman commented:
I updated the UML and classes to address requested changes.

allanhaywood committed:
Updating reaction uml.
